publish-mqtt/src/main.rs

```diff
@@ -363,20 +424,13 @@ async fn connect_start_sensor<'a>(
         .await
         .with_context(|| std::line!().to_string())?;
     match session.start_notify_sensor(&sensor.id).await {
-        Ok(()) => {
-            homie
-                .add_node(sensor.as_node())
-                .await
-                .with_context(|| std::line!().to_string())?;
-            sensor.connection_status = ConnectionStatus::Connected;
-            sensor.last_update_timestamp = Instant::now();
-            Ok(())
-        }
+        Ok(()) => Ok(()),
         Err(e) => {
             // If starting notifications failed a second time, disconnect so
             // that we start again from a clean state next time.
             match sensor.connection_status {
```

Comment:
Is the `connection_status` ever anything other than `Connecting` here?

Reply:
Hrm. I guess not. This suggests that we want to do the retrying here until SENSOR_CONECT_LOCK_TIMEOUT/2 or something and kill off the failed once state? I think this is something I wanted to do anyway eventually.

Reply:
Fixed by moving the `clone_sensor_at_idx` call in `connect_sensor_at_idx` higher up, so it clones the sensor before changing the state of the original to `Connecting`.